setup.py

```diff
@@ -56,6 +56,12 @@
           'schema-salad >= 2.6.20170927145003, < 3',
           'typing >= 3.5.3',
           'six >= 1.8.0',
+          'prov == 1.5.1',
```

Comment:
Does `prov` 1.5.0 work as well?

Reply:
It does but there is a bug with datetime() and that might sometimes not record the start and end time of the activity and many more time instances used in quite a few properties. 
In addition, if you want to make changes to prov serialization later to have a bundle instead of just one prov file (for example in case of sub-workflows), then prov 1.5.1 is better.